cadquery/sketch.py

```diff
@@ -374,9 +365,18 @@ def parray(self: T, r: Real, a1: Real, a2: Real, n: int, rotate: bool = True) ->
             selection = [Vector()]
 
         return self.push(
-            (l * el if isinstance(el, Location) else l * Location(el.Center()))
-            for l in locs
-            for el in selection
+            (
+                l
+                * el
+                * Location(
+                    Vector(0, 0), Vector(0, 0, 1), (a1 + (angle * i)) * int(rotate)
```

Comment:
I'd propose this for readability
```suggestion
                    Vector(0, 0), Vector(0, 0, 1), (a1 + (angle * i)) if rotate else 0
```